src/finchlite/codegen/numba_backend.py

```diff
@@ -716,11 +716,11 @@ def struct_numba_setattr(fmt: AssemblyStructFType, ctx, obj, attr, val):
 
 
 def struct_construct_from_numba(fmt: AssemblyStructFType, numba_struct):
-    args = [
-        construct_from_numba(field_type, getattr(numba_struct, name))
+    kwargs = {
+        name: construct_from_numba(field_type, getattr(numba_struct, name))
         for (name, field_type) in fmt.struct_fields
-    ]
-    return fmt(*args)
+    }
+    return fmt(**kwargs)
```

Comment:
formats now have "multiple constructors", whether we construct from Numba or in the facing API